src/main/java/pl/edu/mimuw/resources/FileInputStreamBufferedReader.java

```diff
@@ -1 +1,40 @@
 package pl.edu.mimuw.resources;
+
+import java.io.BufferedReader;
+import java.io.Closeable;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.CharBuffer;
+
+public class FileInputStreamBufferedReader implements Readable, Closeable {
+    private FileInputStream fileStream;
+    public FileInputStreamBufferedReader(File file) throws IOException {
+        fileStream = new FileInputStream(file);
+    } 
+    public String readText() throws IOException {
+      try(
+        InputStreamReader reader = new InputStreamReader(fileStream);
+        BufferedReader bufferedReader = new BufferedReader(reader);
+      ) {
+          StringBuilder resultString = new StringBuilder();
+          String line = "";
```

Comment:
a może to w końcu to miejsce gdzie przydałoby się nam `do { } while()`? xd bo tak nienaturalnie wygląda inicjalizacja pustym napisem